pypicloud/storage/azure_blob.py

```diff
@@ -64,6 +70,17 @@ def configure(cls, settings):
         if kwargs["storage_account_name"] is None:
             raise ValueError("You must specify the 'storage.storage_account_name'")
 
+        kwargs["storage_account_url"] = settings.get(
+            "storage.storage_account_url", os.getenv("AZURE_STORAGE_SERVICE_ENDPOINT")
+        )
+        if (
+            kwargs["storage_account_url"] is not None
+            and kwargs["storage_account_name"] not in kwargs["storage_account_url"]
+        ):
+            raise ValueError(
+                "You must specify the 'storage.storage_account_name' to match the 'storage.storage_account_url'"
```

Comment:
I added this check because I'm not sure whether you can have a custom private url that doesn't contain the username subdomain